apps/search/js/providers/grid_provider.js

```diff
@@ -21,8 +23,15 @@ DataGridProvider.prototype = {
 
   render: function(results) {
     results.forEach(function(config, index) {
+      if (this.grid.getLength() >= this.MAX_RESULTS) {
```

Comment:
Across providers you can maintain the count in https://github.com/mozilla-b2g/gaia/blob/master/apps/search/js/search.js#L206